htdocs/install/mysql/migration/19.0.0-20.0.0.sql

```diff
@@ -292,3 +292,7 @@ ALTER TABLE llx_receptiondet_batch CHANGE COLUMN fk_commandefourndet fk_elementd
 ALTER TABLE llx_supplier_proposaldet MODIFY ref_fourn VARCHAR(128) NULL;
 
 ALTER TABLE llx_projet ADD COLUMN ref_ext varchar(50) after ref;
+
+-- bookcal
+ALTER TABLE llx_bookcal_availabilities ADD COLUMN entity integer DEFAULT 1 NOT NULL AFTER rowid
```

Comment:
The bookcal_availabilities are linked to a bookcal_calendar (child of bookcal_calendar).
So no need for entity in this table. We only need it into bookcal_calendar.

Reply:
hi @eldy : there might be something I didn't understand in PHP, but I had to add entity field to availabilities table to be able to :
- see (and set) availability on a specific (shared) calendar (=> from ent2, set availability on a ent1 calendar)
- not have warning/error on missing "entity" field in availability_card.php :
 ```
$upload_dir = $conf->bookcal->multidir_output[isset($object->entity) ? $object->entity : 1].'/availabilities';
```
-> I tried without it (trying to modify the less code possible), but trial and errors forced me to do it.


(=>continuing php investigations in the meantime)

Reply:
Dev was initialized by @Hystepik.
I suggest to first submit a PR with the addition of the database structure and only database structure. So adding the new field entity into llx_bookcal_calendar.
Even if not used by code, validation of structure addition is first step to progress...

If a booking calendar is defined into entity X, all availabilities will be saved as linked to booking calendar (so in entity X). 
And if you decide to share the calendar between entities, X and Y, it means you want to share also the availabilities. So no need for entity field into the table of availability.

Reply:
![Screenshot 2024-04-04 200057](https://github.com/Dolibarr/dolibarr/assets/165890649/a9a9418d-c55a-412b-90bd-aea5e5356435)
![Screenshot 2024-04-04 195948](https://github.com/Dolibarr/dolibarr/assets/165890649/5e81e369-2d10-4fa2-8b8a-62577264a92f)


Last screen is from "List calendars", 1st one is from "create availability"